addons/project/static/src/js/project_kanban.js

```diff
@@ -193,6 +272,86 @@ const ProjectTaskKanbanModel = KanbanModel.extend({
             return [old_group.id, new_group.id];
         });
     },
+
+    /**
+     * When grouped by personal stage create a new personal stage instead of
+     * a regular stage.
+     * Meaning setting `user_id` on the stage.
+     *
+     * @override
+     */
+    createGroup: function (name, parentID) {
+        var self = this;
+        var parent = this.localData[parentID];
+        const groupedFieldName = viewUtils.getGroupByField(parent.groupedBy[0]);
+        if (groupedFieldName !== 'personal_stage_type_ids') {
+            return this._super(...arguments);
+        }
+        var groupBy = parent.groupedBy[0];
+        const context = Object.assign({}, parent.context, {
+            default_user_id: self.getSession().user_id[0],
+        });
+        // In case it's a personal stage we don't want to assign it to the project.
+        delete context.default_project_id;
+        return this._rpc({
+                model: 'project.task.type',
+                method: 'name_create',
+                args: [name],
+                context: context,
+            })
+            .then(function (result) {
+                const createGroupDataPoint = (model, parent) => {
+                    const newGroup = model._makeDataPoint({
+                        modelName: parent.model,
+                        context: parent.context,
+                        domain: parent.domain.concat([[groupBy, "=", result[0]]]),
+                        fields: parent.fields,
+                        fieldsInfo: parent.fieldsInfo,
+                        isOpen: true,
+                        limit: parent.limit,
+                        parentID: parent.id,
+                        openGroupByDefault: true,
+                        orderedBy: parent.orderedBy,
+                        value: result,
+                        viewType: parent.viewType,
+                    });
+                    if (parent.progressBar) {
+                        newGroup.progressBarValues = _.extend({
+                            counts: {},
+                        }, parent.progressBar);
+                    }
+                    return newGroup;
+                };
+                const newGroup = createGroupDataPoint(self, parent);
+                parent.data.push(newGroup.id);
+                if (self.isInSampleMode()) {
+                    // in sample mode, create the new group in both models (main + sample)
+                    const sampleParent = self.sampleModel.localData[parentID];
+                    const newSampleGroup = createGroupDataPoint(self.sampleModel, sampleParent);
+                    sampleParent.data.push(newSampleGroup.id);
+                }
+                return newGroup.id;
+            });
```

Comment:
We could use arrow functions and `const` 🙂 
```suggestion
        const parent = this.localData[parentID];
        const groupedFieldName = viewUtils.getGroupByField(parent.groupedBy[0]);
        if (groupedFieldName !== 'personal_stage_type_ids') {
            return this._super(...arguments);
        }
        const groupBy = parent.groupedBy[0];
        const context = Object.assign({}, parent.context, {
            default_user_id: self.getSession().user_id[0],
        });
        // In case it's a personal stage we don't want to assign it to the project.
        delete context.default_project_id;
        return this._rpc({
                model: 'project.task.type',
                method: 'name_create',
                args: [name],
                context: context,
            })
            .then(result => {
                const createGroupDataPoint = (model, parent) => {
                    const newGroup = model._makeDataPoint({
                        modelName: parent.model,
                        context: parent.context,
                        domain: parent.domain.concat([[groupBy, "=", result[0]]]),
                        fields: parent.fields,
                        fieldsInfo: parent.fieldsInfo,
                        isOpen: true,
                        limit: parent.limit,
                        parentID: parent.id,
                        openGroupByDefault: true,
                        orderedBy: parent.orderedBy,
                        value: result,
                        viewType: parent.viewType,
                    });
                    if (parent.progressBar) {
                        newGroup.progressBarValues = _.extend({
                            counts: {},
                        }, parent.progressBar);
                    }
                    return newGroup;
                };
                const newGroup = createGroupDataPoint(this, parent);
                parent.data.push(newGroup.id);
                if (this.isInSampleMode()) {
                    // in sample mode, create the new group in both models (main + sample)
                    const sampleParent = this.sampleModel.localData[parentID];
                    const newSampleGroup = createGroupDataPoint(self.sampleModel, sampleParent);
                    sampleParent.data.push(newSampleGroup.id);
                }
                return newGroup.id;
            });
```